### ISSUE TITLE
Feedback

### ISSUE THREAD

**Issue by tpeczek** [tpeczek/Lib.Net.Http.WebPush#1]
This issue is for general feedback regarding this project.

**Comment by theLeftTenant** [comment]
I read your walk-throughs and this seems like the most well maintained .Net Core Web Push tool. Any interest in strong naming the DLLs?

**Comment by tpeczek** [comment]
Thank you.

Regarding the strong naming, it has positive recommendation and negative aspects are less important those days, so I see no issue with doing that. I've created an issue to track it (#8).

**Comment by vpetrusevici** [comment]
Would be good to add a constructor with no parameters for PushMessage, because with current implementation is impossible to deserailise object of this class from json without creating new class and mapping properties

**Comment by tpeczek** [comment]
@vpetrusevici I will consider that. It opens some error scenarios but maybe it's worth it.

**Comment by rouen-sk** [comment]
Hey, nice library. However, you should probably add `PushServiceClient` constructor accepting IHttpClientFactory (instead of HttpClient) and use `CreateClient` internally to support long-running scenarios correctly.

**Comment by tpeczek** [comment]
Hi @rouen-sk. Thank you for the feedback. In my mind `PushServiceClient` is somethign to be used in various scenarios (think of if like of Refit for example), and accepting `HttpClient` gives the most flexibility. Scenarios like integration with `HttpClientFactory`, in my opinion, should be done from _outside_. That said I'm happy to conider some generic purpose extensions to `HttpClientFactory` which would make this easier (I have some for ASP.NET Core, but from time perspective they seem too opinionated).

**Comment by shareonline** [comment]
I just found this as I am investigating why I can't seem to get the Webpush. Net nuget package to work correctly on android.
I have everything set up and my chromium edge win11 pc gets all notifications correctly, even when deployed to Azure. But I simply can't get android to work with it, and others like progressier for instance work fine.

I was wondering if this library works without issues on newest versions of android/Chrome etc?

And thanks for your work by the way! If this works as expected a beer or alike will certainly be thrown at you :)

Is this updated to use newest versions available for dependencies?

**Comment by tpeczek** [comment]
> I just found this as I am investigating why I can't seem to get the Webpush. Net nuget package to work correctly on android. I have everything set up and my chromium edge win11 pc gets all notifications correctly, even when deployed to Azure. But I simply can't get android to work with it, and others like progressier for instance work fine.
> 
> I was wondering if this library works without issues on newest versions of android/Chrome etc?
> 
> And thanks for your work by the way! If this works as expected a beer or alike will certainly be thrown at you :)
> 
> Is this updated to use newest versions available for dependencies?

Hi @shareonline,

The general answer is yes, this library supports Web Push in all OS/browser scenarios where the browser supports it. This includes Android/Chrome.

One Chrome specific aspect that has surfaced with one of latest versions, is that Chrome has started to complain about certain VAPID keys, you can read more in this issue: #29. If that is your case, than you need to play a little bit with keys provided in configuration. If you are picking up any other error, please share it and I'll take a look as soon as I will be able to carve some time.

**Comment by vasilp** [comment]
Can this library specify icon and URL to navigate to when the user clicks on the notificaiton?

**Comment by tpeczek** [comment]
> Can this library specify icon and URL to navigate to when the user clicks on the notificaiton?

Hi @vasilp,

For Web Push notifications, icon and URL are a client side thing. Depending on the type of client there are different ways to control them. For example in case of browser there are parameters to `showNotification` call and you can provide the values from anywhere, including the payload from the server. Some clients (icluding some ready to use service workers for browser) will automatically pick this stuff if the payload coming from the server is of specific format. As this library is completely agnostic when it comes to payload format, it supports all the above scenarios.

For example, this past question describes using this library with Angular built-in service worker and provides more details on the above: https://github.com/tpeczek/Lib.Net.Http.WebPush/issues/10

**Comment by angelru** [comment]
It can be used in a Blazor app, right?

**Comment by tpeczek** [comment]
Hi @angelru,

> It can be used in a Blazor app, right?

On the server-side, certainly.

On the client-side, technically maybe but I have never attempted to test this. That is because that would be an attempt to achieve quite an unusual use case (in fact potentially not a one even desired to be handled by Web Push in general) and bring a security risk (all keys would need to end up on the frontend where they can be relatively easily sniffed and misused).

**Comment by angelru** [comment]
> Hi [@angelru](https://github.com/angelru),
> 
> > It can be used in a Blazor app, right?
> 
> On the server-side, certainly.
> 
> On the client-side, technically maybe but I have never attempted to test this. That is because that would be an attempt to achieve quite an unusual use case (in fact potentially not a one even desired to be handled by Web Push in general) and bring a security risk (all keys would need to end up on the frontend where they can be relatively easily sniffed and misused).

I have a Blazor WASM application with a backend for a frontend (BFF), I understand that it's possible, right?

**Comment by tpeczek** [comment]
> I have a Blazor WASM application with a backend for a frontend (BFF), I understand that it's possible, right?

Yes, the library should be used in your backend service (exposed via an API endpoint that will be called from the WASM part).

**Comment by shareonline** [comment]
@tpeczek Just wanted to let you know i am using this with great success and stability for a long time now 🥇 
Thank you for your contribution here!

For everyone else, if people are having issues with edge browser on android, its a Microsoft problem and have been there for a long time now. Probably never gonna get fixed, so maybe use something else.. All other browsers seem to work perfectly 🚀

Oh and using it from backend in a blazor server application

**Comment by tpeczek** [comment]
Thank you for the kind words @shareonline and happy to hear that it works good for you 😊
